pyiceberg/catalog/glue.py

```diff
@@ -303,11 +303,17 @@ def add_glue_catalog_id(params: Dict[str, str], **kwargs: Any) -> None:
 
 
 class GlueCatalog(MetastoreCatalog):
-    def __init__(self, name: str, **properties: Any):
+    glue: GlueClient
+
+    def __init__(self, name: str, client: GlueClient | None = None, **properties: Any):
         super().__init__(name, **properties)
 
         retry_mode_prop_value = get_first_property_value(properties, GLUE_RETRY_MODE)
 
+        if client:
+            self.glue = client
+            return
```

Comment:
Nit! Can we turn this into an `if ... else`? I think the `return` is harder to read.